pkg/controller/directvolumemigration/task.go

```diff
@@ -409,77 +411,31 @@ func (t *Task) Run() error {
 			}
 			t.Requeue = PollReQ
 		}
-	case CreatePVProgressCRs:
-		err := t.createPVProgressCR()
+	case RunRsyncOperations:
+		allCompleted, anyFailed, failureReasons, err := t.runRsyncOperations()
 		if err != nil {
 			return liberr.Wrap(err)
 		}
-		t.Requeue = NoReQ
-		if err = t.next(); err != nil {
-			return liberr.Wrap(err)
+		t.Requeue = PollReQ
+		if allCompleted {
+			t.Requeue = NoReQ
+			if anyFailed {
+				t.fail(MigrationFailed, failureReasons)
+				return nil
+			}
+			if err = t.next(); err != nil {
+				return liberr.Wrap(err)
+			}
 		}
-	case CreateRsyncClientPods:
-		err := t.createRsyncClientPods()
+	case CreatePVProgressCRs:
+		err := t.createPVProgressCR()
 		if err != nil {
 			return liberr.Wrap(err)
 		}
 		t.Requeue = NoReQ
 		if err = t.next(); err != nil {
 			return liberr.Wrap(err)
 		}
-	case WaitForRsyncClientPodsCompleted:
-		completed, failed, err := t.haveRsyncClientPodsCompletedOrFailed()
-		if err != nil {
-			return liberr.Wrap(err)
-		}
-		if completed {
-			isStunnelTimeout, err := hasAllRsyncClientPodsTimedOut(t.getPVCNamespaceMap(), t.Client, t.Owner.Name)
```

Comment:
These functions are now called directly in the RunRsyncOperations phase to keep this switch case minimal